python/ray/experimental/workflow/workflow_context.py

```diff
@@ -55,6 +55,11 @@ def update_workflow_step_context(context: Optional[WorkflowStepContext],
     global _context
     _context = context
     _context.workflow_scope.append(step_id)
+    # avoid cyclic import
```

Comment:
Usually, when this happens, it means something in design is wrong.

Reply:
I think we need a "workflow" abstract base class. This addresses most issues.